Fixes for defects found by coverity scan.

runrevmark opened the pull request.

runrevmark committed:
[WarningFix] Corrected EOF to IO_EOF in MCFilesExecPerformReadCodeUnit.

runrevmark committed:
[CI 17043] engine: Make sure t_bytes is used to build the UTF-32 string in MCFilesExecPerformReadCodeUnit.

runrevmark committed:
[CI 16865] engine: Fix resource leaks and error handling in MCFilesExecPerformReadFixedFor.

runrevmark committed:
[CI 16130] engine: Potential uninitialized value (t_stat) in MCFilesExecWriteToStdout.
[CI 16121] engine: Potential uninitialized value (t_stat) in MCFilesExecWriteToStderr.

Issues fixed by ensuring out paramete r_stat is set on all codepaths in MCFilesExecWriteToStream.

runrevmark committed:
[CI 17039] libgraphics: Ensure non-rectangular code path in MCGRegionCopyWithTransform sets r_copy.

runrevmark committed:
[WarningFix] Remove extraneous ! operator in search for mouse up event in MCEventQueueGetMouseClick.

runrevmark committed:
[CI 17033] engine: Fix use of potentially uninitalized t_mouse_move pointer in MCEventQueueGetMouseClick().

runrevmark committed:
[CI 17018] engine: stackcache: Make sure upper capacity index does not exceed the number of entries in __kMCValueHashTableCapacities in RehashBuckets().
[CI 16824] engine: stackcache: Make sure we don't continue if we run out of capacity indicies in RehashBuckets().
[CI 16978] <ditto>

runrevmark committed:
[CI 16766] engine: stackcache: Dead code caused by signed / unsigned comparison.
[CI 14904] <ditto>

Fixed by changing signature of RehashBuckets() to take a signed value - item_count_delta.

runrevmark committed:
[CI 17028] engine: deploy: Fix resource leak in MCIdeDmgBuild::exec_ctxt (replace array ptr with MCAutoArray).

runrevmark committed:
[CI 16027] engine: deploy: Use of potentially uninitialized variable t_data_size.
[CI 15954] engine: deploy: Use of potentially uninitialized variable t_stat.

Fixed by ensuring an error on the context causes an immediate return.

runrevmark committed:
[CI 17007] engine: deploy: Potential use of failed fopen in MCIdeDmgDump.

runrevmark committed:
[CI 17006] engine: deploy: m_platform not initialized in MCIdeDiet.
[CI 16780] engine: deploy: m_platform not initialized in MCIdeSign.
[CI 16336] engine: deploy: m_platform not initialized in MCIdeDeploy.

Fixed by adding PLATFORM_NONE to the relevant enums and adding appropriate initializers in the constructors.

runrevmark committed:
[CI 15444] engine: deploy: Replace t_module_files with MCAutoArray to prevent NULL dereference.

runrevmark committed:
[CI 15275] engine: deploy: Replace t_aux_stackfiles with MCAutoArray to prevent NULL dereference.

runrevmark committed:
[CI 16822] engine: deploy: Do nothing in add_version_info_entry if no context specified.

runrevmark committed:
[CI 16182] engine: deploy: Fix memory leak for generated bytes array in add_version_info_entry.

runrevmark committed:
[CI 16236] engine: deploy: Fix potential use of uninitialized field due to not checking for success in MCDeploySignLoadPVK.

runrevmark committed:
[CI 15373] engine: deploy: Make sure we only zero signature field in t_request -> blob if non-nil in MCDeploySignWindowsAddTimeStamp.

runrevmark committed:
[CI 15963] engine: deploy: Make sure success is checked before reading fields from read structs in MCDeployToELF.

runrevmark committed:
[CI 14843] engine: deploy: Use 'free' rather than delete in MCDeployToLinuxReadString.

runrevmark committed:
[CI 15169] engine: deploy: Move more code into ifdef 0 in MCDeployDmgBuild.

runrevmark committed:
[CI 17031] engine: field: Do nothing if non scroll action passed to MCField::fscroll.

peter-b commented on engine/src/deploy.cpp:
```diff
@@ -973,19 +974,20 @@ void MCIdeExtract::exec_ctxt(MCExecContext& ctxt)
     if (!ctxt . EvalExprAsStringRef(m_filename, EE_IDE_EXTRACT_BADFILENAME, &t_filename))
         return;
 		
+	if (!ctxt . HasError())
```
Is the sense of this test reversed?  Should the test be: 

```
if (ctxt . HasError())
```

runrevmark replied:
Its not even necessary at all - if there is an error at that point, then the function would have already returned!

runrevmark committed:
[CID 16027] Revision

Remove unnecessary HasError() check. At the point 'HasError()' is checked there can be
no errors as any of the preceeding code in the handler would have already returned if
there were.

runrevmark committed:
Merge branch 'develop' into coverity

runrevmark committed:
[CID 16827] engine: util.cpp: Failure to free result of MCStringConvertToCString.

Fixed by using an MCAutoPointer around the allocated data.

runrevmark committed:
[CID 16624] engine: util.cpp: Failure to free result of MCStringConvertToCString in MCU_parsepoint.

Fixed by using an MCAutoPointer around the allocated data.

peter-b commented on engine/src/deploy.cpp:
```diff
@@ -334,12 +336,10 @@ bool MCDeployWriteCapsule(const MCDeployParameters& p_params, MCDeployFileRef p_
 
 	MCDeployCapsuleDestroy(t_capsule);
-	for(uint32_t i = 0; i < MCArrayGetCount(p_params.auxillary_stackfiles); i++)
+	for(uint32_t i = 0; i < t_aux_stackfiles . Size(); i++)
```
`MCAutoArray<>::Size()` returns a value of type `uindex_t`, not `uint32_t`.

runrevmark replied:
Fixed that instance, and the similar ones in the same function.

peter-b commented on engine/src/deploy.cpp:
```diff
@@ -836,20 +842,15 @@ void MCIdeDmgBuild::exec_ctxt(MCExecContext& ctxt)
 
 	if (!ctxt . HasError())
-		for(uint32_t i = 0; i < t_item_count && !ctxt . HasError(); i++)
+		for(uint32_t i = 0; i < t_items . Size() && !ctxt . HasError(); i++)
```
`uindex_t` loop variable

runrevmark replied:
Fixed.

peter-b commented on engine/src/deploy_sign.cpp:
```diff
@@ -1034,7 +1034,8 @@ static bool MCDeploySignWindowsAddTimeStamp(const MCDeploySignParameters& p_para
 	if (t_request != nil)
 	{
-		t_request -> blob -> signature = nil;
+        if (t_request -> blob != nil)
```
This would probably be easier to read if you combined the `if` conditions, i.e.:

```
if (t_request != nil && t_request->blob != nil)
    SpcTimeStampRequest_free(t_request);
```

runrevmark replied:
That's not quite right. If t_request is not nil it needs to be freed. If it has a non-nil blob field it needs its signature unhooked as that's borrowed from elsewhere.

I've clarified this behavior with comments.

peter-b replied:
Ah - it turns out that I was reading the code incorrectly because of mixed tab/space indenting. Sorry for wasting your time. :disappointed:

runrevmark committed:
[[ Cleanup ]] Make sure for loop iterand variable matches type of conditions in MCDeployWriteCapsule.

runrevmark committed:
[[ Cleanup ]] Make sure for loop iterand variable matches type of conditions in MCIdeDmgBuild::exec_ctxt.

runrevmark committed:
[[ Cleanup ]] Clarified ownership of 'signature' field in SpcTimeStampRequest in MCDeploySignWindowsAddTimestamp.

peter-b commented:
Reviewed-By: Peter TB Brett peter@livecode.com
